Source/SharpNeedle/HedgehogEngine/Mirage/SplinePath.cs

```diff
@@ -0,0 +1,283 @@
+        writer.Write(0x200); // Version?
+
+        writer.WriteObject(new BinaryList<PathObject>(Paths));
+    }
+}
+
+public class PathObject : IBinarySerializable
+{
+    public string Name { get; set; }
+    public bool Field04 { get; set; }
+    public bool Field05 { get; set; }
+    public float Distance { get; set; }
+    public int Field48 { get; set; }
+    public AABB Aabb { get; set; }
+    public UnknownStruct Unknown { get; set; }
+    public List<bool> KnotFlags { get; set; } = new();
+    public List<float> KnotDistances { get; set; } = new();
+    public List<Vector3> Knots { get; set; } = new();
+    public List<Vector3> KnotTangentIns { get; set; } = new();
+    public List<Vector3> KnotTangentOuts { get; set; } = new();
+    public List<DoubleKnot> DoubleKnots { get; set; } = new();
+    public List<UserData> UserDatas { get; set; } = new();
+
+    public void Read(BinaryObjectReader reader)
+    {
+        Name = reader.ReadStringOffset();
+        Field04 = reader.Read<bool>();
+        Field05 = reader.Read<bool>();
+
+        short knotCount = reader.Read<short>();
+
+        Distance = reader.Read<float>();
+
+        KnotFlags.AddRange(reader.ReadArrayOffset<bool>(knotCount));
+        KnotDistances.AddRange(reader.ReadArrayOffset<float>(knotCount));
+        Knots.AddRange(reader.ReadArrayOffset<Vector3>(knotCount));
+        KnotTangentIns.AddRange(reader.ReadArrayOffset<Vector3>(knotCount));
+        KnotTangentOuts.AddRange(reader.ReadArrayOffset<Vector3>(knotCount));
+
+        DoubleKnots.AddRange(reader.ReadObjectArrayOffset<DoubleKnot>(reader.Read<int>() / 2));
+
+        Aabb = reader.Read<AABB>();
+
+        UserDatas.AddRange(reader.ReadObjectArrayOffset<UserData>(reader.Read<int>()));
+
+        Field48 = reader.Read<int>(); // Always 0?
+
+        Unknown = reader.ReadObjectOffset<UnknownStruct>();
+    }
+
+    public void Write(BinaryObjectWriter writer)
+    {
+        writer.WriteStringOffset(StringBinaryFormat.NullTerminated, Name);
+        writer.Write(Field04);
+        writer.Write(Field05);
+
+        writer.Write((short)Knots.Count);
+
+        writer.Write(Distance);
+
+        writer.WriteArrayOffset(KnotFlags.ToArray(), 4);
+        writer.WriteArrayOffset(KnotDistances.ToArray(), 4);
+        writer.WriteArrayOffset(Knots.ToArray(), 4);
+        writer.WriteArrayOffset(KnotTangentIns.ToArray(), 4);
+        writer.WriteArrayOffset(KnotTangentOuts.ToArray(), 4);
+
+        writer.Write(DoubleKnots.Count * 2);
+        if (DoubleKnots.Count != 0)
+            writer.WriteObjectCollectionOffset(DoubleKnots);
+        else
+            writer.Write<int>(0);
+
+        writer.Write(Aabb);
+
+        writer.Write(UserDatas.Count);
+        writer.WriteObjectCollectionOffset(UserDatas);
+
+        writer.Write(Field48);
+
+        writer.WriteObjectOffset(Unknown);
+    }
+
+    public override string ToString()
+    {
+        return Name;
+    }
+}
+
+public class DoubleKnot : IBinarySerializable
```

Comment:
Is this required to be a class? It should work just fine as a struct without IBinarySerializable.

Reply:
Resolved in 6a814e4.